### ISSUE TITLE
[help wanted] run server streaming

### ISSUE THREAD

**Issue by alidabaghi** [k2-fsa/sherpa#238]
hello, everyone. I have been installing sherpa and when I wanted to run streaming server.py. encounter with this error, 

![Uploading image.png…]()

**Comment by csukuangfj** [comment]
Could you re-upload the image? It is not readable.

**Comment by e871** [comment]
ok. thanks for  concern
![Screenshot from 2022-12-06 12-30-47](https://user-images.githubusercontent.com/78419335/205868267-5ff44e1e-7fba-4e01-aefe-24109d42360c.png)

**Comment by e871** [comment]
I have checked icefall files and found pt model.

![Screenshot from 2022-12-06 12-43-03](https://user-images.githubusercontent.com/78419335/205869448-3a160145-f65b-4a87-97b0-bd4ea3354a91.png)

**Comment by e871** [comment]
do i downloaded from another repository???

**Comment by csukuangfj** [comment]
you can see that the `.pt` file is only 134 bytes

Please use
```
cd exp
git lfs pull --include "cpu-jit-epoch-30-avg-10-torch-1.6.0.pt"
```

(I think you have missed the command `git lfs install` as described in the documentation)

**Comment by e871** [comment]
thank you, i am downloading.

**Comment by e871** [comment]
hello again, I downloaded the model and  encounter this error
![Screenshot from 2022-12-06 14-58-39](https://user-images.githubusercontent.com/78419335/205900236-685adbb2-61cd-4e06-932c-4dfb87a9e6b3.png)

**Comment by e871** [comment]
can you help me, again

**Comment by csukuangfj** [comment]
Are you using the latest version of sherpa? If not, please use the latest version of sherpa.

**Comment by e871** [comment]
I installed this version\
![Screenshot from 2022-12-06 15-07-32](https://user-images.githubusercontent.com/78419335/205901806-bbf0dc27-9861-4c7b-86b8-7a4088e91960.png)

**Comment by csukuangfj** [comment]
> I installed this version ![Screenshot from 2022-12-06 15-07-32](https://user-images.githubusercontent.com/78419335/205901806-bbf0dc27-9861-4c7b-86b8-7a4088e91960.png)
> 
> can you help me, again

We have been in `1.0` for a long time. It is not clear which commit you are using.

How did you install sherpa?

What is the output of 
```python3
sherpa-version
```

The line numbers in your error logs don't match the latest file
https://github.com/k2-fsa/sherpa/blob/master/sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py

**Comment by e871** [comment]
![Screenshot from 2022-12-06 15-28-39](https://user-images.githubusercontent.com/78419335/205905911-6314894c-1f7f-4e09-9e84-969b4277c0b3.png)

**Comment by csukuangfj** [comment]
How did you install sherpa?

**Comment by e871** [comment]
with pip

**Comment by csukuangfj** [comment]
> ![Screenshot from 2022-12-06 15-28-39](https://user-images.githubusercontent.com/78419335/205905911-6314894c-1f7f-4e09-9e84-969b4277c0b3.png)

The git sha1 and git date are empty. It is not possible to tell which commit you are using.

---

Please install the latest master of sherpa.

**Comment by e871** [comment]
i used this page for install sherpa 
https://k2-fsa.github.io/sherpa/python/installation/index.html

**Comment by e871** [comment]
thank you. i am trying installl again

**Comment by csukuangfj** [comment]
Remember to uninstall `sherpa` first
```
pip uninstall k2-sherpa
```

**Comment by e871** [comment]
I am installing. With 96  mycomputer locked 
![16703309835606896063639228848608](https://user-images.githubusercontent.com/78419335/205917168-0b330be9-d68b-4684-b9dc-35af977d41f4.jpg)

**Comment by e871** [comment]
I am waiting long time even now

**Comment by e871** [comment]
![Screenshot from 2022-12-06 17-13-18](https://user-images.githubusercontent.com/78419335/205928539-c6ad0000-981c-4bf7-8081-afba4743d1b8.png)

**Comment by e871** [comment]
i forced restart my pc

**Comment by csukuangfj** [comment]
Could you change
https://github.com/k2-fsa/sherpa/blob/b933ac22b6f1d1871d10d52bc2ee60ea0b115f3b/cmake/cmake_extension.py#L104
to
```
-j1
```

before running `python3 setup.py install`?

**Comment by e871** [comment]
thank you for replying again. i am installing now

**Comment by e871** [comment]
is that true???

![Screenshot from 2022-12-06 17-30-58](https://user-images.githubusercontent.com/78419335/205932378-2bbb2165-2b6d-448b-937a-c64dfdd08577.png)

**Comment by csukuangfj** [comment]
Did you use `wget` or `git clone` to download the source code?
It's strange that `git sha1` and `git date` is empty for sherpa.

Can you re-run your `streaming_server.py` and see if it still causes errors?

**Comment by e871** [comment]
hello agian. i have downloded sherpa from master and succesfully installed web server. 
i am very thrilling in speed of decoding.  
 do i read how to improving speed of decoding???
can i replace my model????

**Comment by csukuangfj** [comment]
> do i read how to improving speed of decoding???

Do you have a problem with the speed of decoding? Does it run very slowly?

----

> can i replace my model????

What is the type of your model? Transducer, ctc, or attention based encoder-decoder, or any other kinds of models?

---

> can i train this models with mexican language???

Yes, as long as you have the data and some GPUs. The training code is at http://github.com/k2-fsa/icefall
This repo is for speech recognition with pre-trained models from icefall.

**Comment by e871** [comment]
> > do i read how to improving speed of decoding???
> 
> Do you have a problem with the speed of decoding? Does it run very slowly?
> 
> > can i replace my model????
> 
> What is the type of your model? Transducer, ctc, or attention based encoder-decoder, or any other kinds of models?
> 
> > can i train this models with mexican language???
> 
> Yes, as long as you have the data and some GPUs. The training code is at [http://github.com/k2-fsa/icefall](https://github.com/k2-fsa/icefall) This repo is for speech recognition with pre-trained models from icefall.

no i mean, very fast.

**Comment by e871** [comment]
> 
I have trained conformer with nemo toolkit.

**Comment by csukuangfj** [comment]
> I have trained conformer with nemo toolkit.


What kind of loss are you using? 

CTC loss ?

**Comment by e871** [comment]
> 

yes

**Comment by csukuangfj** [comment]
> > 
> 
> yes

Then it is supported by sherpa.

Are you able to export your model via torchscript, either with `torch.jit.script()` or `torch.jit.trace()`?

If so, are yo able to make it public? (If so, we can add it to sherpa. Otherwise, we can tell you how to modify the code and you have to make the changes by yourself).

**Comment by e871** [comment]
> 

I have pretraind conformer bpe model and use ctc loss. i use pyctc for decoding.

**Comment by e871** [comment]
> 

my model  havent high quality now. and  i am trying improve model accuracy. then i public model

**Comment by csukuangfj** [comment]
If you are going to make the model public, is it possible to provide a model for testing so that I can add it to sherpa.
(The model does not need to be of high recognition accuracy).

You only need to provide a torchscript model and a token list (mapping token to its integer ID).

**Comment by e871** [comment]
I have trained this model from this link
https://github.com/NVIDIA/NeMo/tree/b8cc5030d4cb95a7e3ce7a8c22ef9d736c5c0bfc/nemo/collections/asr/models

**Comment by e871** [comment]
i have trained squeez former

**Comment by csukuangfj** [comment]
> https://github.com/NVIDIA/NeMo/tree/b8cc5030d4cb95a7e3ce7a8c22ef9d736c5c0bfc/nemo/collections/asr/models

Thanks!

Are you able to provide a torchscript model?

**Comment by csukuangfj** [comment]
You may find
https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/core/export.html
useful.

**Comment by e871** [comment]
thank you.  i will be trying load model in torch script.

**Comment by e871** [comment]
i may be trying to  train your model with Mexican language????

**Comment by csukuangfj** [comment]
> i may be trying to train your model with Mexican language????

Yes, that is also possible. How large is your dataset (i.e., how many hours of your training data)?

---

If you already have a pre-trained model, it may be easier to export it via torchscript and use it with sherpa.
It may take some time to train a model.

please  have a look at http://github.com/k2-fsa/icefall
if you want to train your model with icefall. Feel free to create any issues if you encounter problems.

**Comment by e871** [comment]
can i help you to how preproccesing data and train it???

**Comment by csukuangfj** [comment]
Could you describe what is the format of your data?

Please have a look at https://github.com/lhotse-speech/lhotse
for how to prepare your data.

**Comment by csukuangfj** [comment]
> > 
> 
> I have trained conformer with nemo toolkit.

@alidabaghi 

You may find https://github.com/k2-fsa/sherpa/pull/332 useful.

We now support pre-trained CTC models from NeMo.
